### PR TITLE
contributing: added contributing guidlines

### PR DESCRIPTION
**Pull request by LeanderK**
added a contributing.md to aid with contributing to our project

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+also how to reproduce the bug.
+
+# Making Changes
+ * Create a topic branch from where you want to base your work.
+ * Make commits of logical units.
+ * Check for unnecessary whitespace with git diff --check before committing.
+ * Make sure your commit messages are readable and understandable.
+ * Make sure you have added the necessary tests for your changes.
+ * Run all the tests to assure nothing else was accidentally broken.
+ * If a commit fixed a bug, include `fixes #IUSSUE` in the message (more information can be found [here](https://help.github.com/articles/closing-issues-via-commit-messages/))
```
Comment (kelunik) [inline CONTRIBUTING.md]: `fixes #ISSUE` < typo

Comment (LeanderK) [inline CONTRIBUTING.md]: Where is the typo?

**Comment by kelunik** [comment]
Can be merged after the typo is fixed.

File: CONTRIBUTING.md (hunk 1 above)
Comment (kelunik) [inline CONTRIBUTING.md]: `fixes #IUSSUE` → `fixes #ISSUE`

Comment (LeanderK) [inline CONTRIBUTING.md]: ah sry i thought the comment were the typo and not the corrected version

Comment (LeanderK) [inline CONTRIBUTING.md]: fixed

Comment (kelunik) [inline CONTRIBUTING.md]: yes, was unclear, sorry.

**Pull request merged**